opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -87,21 +87,34 @@ def detect(self) -> "Resource":
 
 
 def get_aggregated_resources(
-    detectors: typing.List["ResourceDetector"], initial_resource=None,
+    detectors: typing.List["ResourceDetector"],
+    initial_resource: typing.Optional[Resource] = None,
+    timeout=5,
 ) -> "Resource":
+    """ Retrieves resources from detectors in the order that they were passed
+
+    :param detectors: List of resources in order of priority
+    :param initial_resource: Static resource. This has highest priority
+    :param timeout: Number of seconds to wait for each detector to return
+    :return:
+    """
     final_resource = initial_resource or _EMPTY_RESOURCE
-    final_resource = final_resource.merge(OTELResourceDetector().detect())
-    token = attach(set_value("suppress_instrumentation", True))
-    for detector in detectors:
-        try:
-            detected_resources = detector.detect()
-        # pylint: disable=broad-except
-        except Exception as ex:
-            if detector.raise_on_error:
-                raise ex
-            logger.warning("Exception in detector %s, ignoring", detector)
-            detected_resources = _EMPTY_RESOURCE
-        finally:
-            final_resource = final_resource.merge(detected_resources)
-    detach(token)
+    detectors = [OTELResourceDetector()] + detectors
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Does this run the detectors in order? We have to preserve the ordering since the earlier detectors take priority right?

Reply:
While it doesn't run the detectors in order, it does gather the results in order, so yeah priority is maintained. There are also unit tests checking for this.